shoes-core/lib/shoes/link.rb

```diff
@@ -62,6 +62,17 @@ def visible?
       text_block_guard && @text_block.visible?
     end
 
+    def links
+      [self]
+    end
+
+    def self.find_links(texts)
+      texts.to_a
+           .map { |text| text.links if text.respond_to?(:links) }
+           .flatten
+           .compact
```

Comment:
maybe select responds_to links first? That way we'd get rid of the if and most likely the compact as well?